cmd/root_test.go

```diff
@@ -2141,6 +2142,79 @@ func TestProjectRootIsSymlink(t *testing.T) {
 	)
 }
 
+func TestConcurrentInvocation(t *testing.T) {
+	as := require.New(t)
+
+	tempDir := test.TempExamples(t)
+	configPath := filepath.Join(tempDir, "/treefmt.toml")
+
+	test.ChangeWorkDir(t, tempDir)
+
+	cfg := &config.Config{
+		FormatterConfigs: map[string]*config.Formatter{
+			"echo": {
+				Command:  "echo",
+				Includes: []string{"*"},
+			},
+			"slow": {
+				Command: "test-fmt-delayed-append",
+				// connect timeout for the db is 1 second
+				// wait 2 seconds before appending ' ' to each provided path
+				Options:  []string{"2", " "},
+				Includes: []string{"*"},
+			},
+		},
+	}
+
+	eg := errgroup.Group{}
+
+	// concurrent invocation with one slow instance and one not
+
+	eg.Go(func() error {
+		treefmt(t,
+			withArgs("--formatters", "slow"),
+			withConfig(configPath, cfg),
+			withNoError(t),
+		)
+
+		return nil
+	})
+
+	time.Sleep(time.Second)
+
+	treefmt(t,
+		withArgs("--formatters", "echo"),
+		withConfig(configPath, cfg),
+		withError(func(as *require.Assertions, err error) {
+			as.ErrorContains(err, "failed to open cache")
+		}),
+	)
+
+	as.NoError(eg.Wait())
+
+	// concurrent invocation with one slow instance and one configured to clear the cache
+
+	eg.Go(func() error {
+		treefmt(t,
+			withArgs("--formatters", "slow"),
+			withConfig(configPath, cfg),
+			withNoError(t),
+		)
+
+		return nil
+	})
+
+	time.Sleep(time.Second)
+
+	treefmt(t,
+		withArgs("-c", "--formatters", "echo"),
```

Comment:
nit, feel free to ignore:

```suggestion
		withArgs("--clear-cache", "--formatters", "echo"),
```